From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Wed, 15 Nov 2017 23:02:34 +0100
Subject: [PATCH] {BUG in 4.15-rc1} drivers/mmc/core: initialize drv_type

clang reports:

    drivers/mmc/core/mmc.c:1303:6: error: variable 'drv_type' is used
    uninitialized whenever 'if' condition is true
          [-Werror,-Wsometimes-uninitialized]
            if (fixed_drv_type >= 0)
                ^~~~~~~~~~~~~~~~~~~
    drivers/mmc/core/mmc.c:1313:6: note: uninitialized use occurs here
            if (drv_type)
                ^~~~~~~~
    drivers/mmc/core/mmc.c:1303:2: note: remove the 'if' if its
    condition is always false
            if (fixed_drv_type >= 0)
            ^~~~~~~~~~~~~~~~~~~~~~~~
    drivers/mmc/core/mmc.c:1297:45: note: initialize the variable
    'drv_type' to silence this warning
            int card_drv_type, drive_strength, drv_type;
                                                       ^
                                                        = 0
---
 drivers/mmc/core/mmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/core/mmc.c b/drivers/mmc/core/mmc.c
index 588482ddb281..811fffef8217 100644
--- a/drivers/mmc/core/mmc.c
+++ b/drivers/mmc/core/mmc.c
@@ -1294,7 +1294,7 @@ int mmc_hs400_to_hs200(struct mmc_card *card)
 
 static void mmc_select_driver_type(struct mmc_card *card)
 {
-	int card_drv_type, drive_strength, drv_type;
+	int card_drv_type, drive_strength, drv_type = 0;
 	int fixed_drv_type = card->host->fixed_drv_type;
 
 	card_drv_type = card->ext_csd.raw_driver_strength |
-- 
